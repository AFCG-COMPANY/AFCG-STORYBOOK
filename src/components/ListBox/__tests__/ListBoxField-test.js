import React from 'react';
import { mount } from 'enzyme';
import ListBox from '../';

describe('ListBoxField', () => {
  it('should render', () => {
    const wrapper = mount(
      <ListBox.Field>
        <ListBox.Selection clearSelection={jest.fn()} />
      </ListBox.Field>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should be focusable', () => {
    const wrapper = mount(
      <ListBox.Field>
        <ListBox.Selection clearSelection={jest.fn()} />
      </ListBox.Field>
    );
    expect(wrapper.children().prop('tabIndex')).toBe('0');
  });
});
