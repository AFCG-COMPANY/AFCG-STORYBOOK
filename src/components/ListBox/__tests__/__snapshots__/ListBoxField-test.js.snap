// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBoxField should render 1`] = `
<ListBoxField>
  <div
    className="bx--list-box__field"
    role="button"
    tabIndex="0"
  >
    <ListBoxSelection
      clearSelection={[MockFunction]}
      translateWithId={[Function]}
    >
      <div
        className="bx--list-box__selection"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
        title="Clear selected item"
      >
        <Icon
          description="Clear selected item"
          fillRule="evenodd"
          focusable="false"
          name="close"
          role="img"
        >
          <svg
            alt="Clear selected item"
            aria-label="Clear selected item"
            fillRule="evenodd"
            focusable="false"
            height="10"
            name="close"
            role="img"
            viewBox="0 0 10 10"
            width="10"
          >
            <title>
              Clear selected item
            </title>
            <path
              d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
              key="key0"
            />
          </svg>
        </Icon>
      </div>
    </ListBoxSelection>
  </div>
</ListBoxField>
`;
